### PR TITLE
doc: net: Add networking changes to 2.1 release note

### PR DESCRIPTION
**Pull request by jukkar**
Initial set of networking changes in 2.1 release.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Comment by rlubos** [comment]
We could also mention the following OpenThread updates:
* Updated the supported version of OpenThread (#20106),
* Updated OpenThread configuration, to use mbedTLS provided by Zephyr (#15697, PR was quite old, but merged just recently)

File: doc/releases/release-notes-2.1.rst [hunk 1]
```diff
@@ -36,7 +36,51 @@ Drivers and Sensors
...
 
-* TBD
+* Added new TCP stack implementation. The new TCP stack is still experimental
+  and is turned off by default. Users wanting to experiment with it can set
+  :option:`CONFIG_NET_TCP2` Kconfig option.
+* Added support for running MQTT protocol on top of a Websocket connection.
+* Added infrastructure for testing network samples in automatic way.
+* Added support for enabling DNS in LWM2M.
+* Added support for resetting network statistics in net-shell.
+* Added support for getting statistics how long time it took to receive or send
```
Comment (dbkinder) [inline doc/releases/release-notes-2.1.rst]: Better would be:
```
* Added support for getting statistics about the time it took to receive or send
  a network packet.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by jukkar** [comment]
Updated according to comments.

**Review by dbkinder — APPROVED** [review]
LGTM, thanks.

**Pull request merged**